enhancements/microshift/update-and-rollback.md

```diff
@@ -0,0 +1,180 @@
+rollback mechanism is in place in case something goes wrong.
+
+### Goals
+
+The goals of this proposal are:
+- How to implement a successful MicroShift upgrade on systems based on `rpm-ostree`
+- Describe the process of creating a MicroShift's data backup
+- Identify what files need to be saved as part of a MicroShift's data backup
+- Describe the process of rolling back to a previous working state
+
+### Non-Goals
+
+The non-goals of this proposal are:
+- How to implement MicroShift healthchecks using `greenboot`
+- How to implement upgrades of 3rd party applications running on MicroShift
+- How to implement rollbacks of 3rd party applications running on MicroShift
+
+## Proposal
+
+### Workflow Description
+
+The Operating System of an Edge Device running MicroShift will be created using
+Image Builder. This tool will allow `device administrators` to customize an edge
+optimized version of RHEL by adding RPMs and configuration of choice as described
+[here](https://github.com/openshift/enhancements/blob/master/enhancements/microshift/kubernetes-for-device-edge.md).
+
+When a new version of MicroShift needs to be rolled out, a new image shall be created
+and exposed to the edge devices. When working with thousands of Edge Devices,
+automatic detection and enforcement of upgrades is required.  Hence, Edge Devices will
+automatically detect the new exposed OSTree and perform a transactional upgrade.
+
+Before pivoting to the new Operating System Tree, a data backup needs to be performed
+in order to save the state of the running Edge Device. Due to the resource constrained
+environments where these Edge Devices run on, only the N-1 state will be supported. That 
+means that only one backup will be stored and when the upgrade is performed, the Edge
+Device can only rollback to the previous state N-1.
+
+The backup should consider all data required to save the state of MicroShift such as
+certificates, configuration files, volumes
+
+At every boot, `Greenboot` healthchecks will be executed as described in [this enhancement.](https://github.com/openshift/enhancements/blob/master/enhancements/microshift/microshift-greenboot.md)
+
+If upgrading MicroShift to a new version results in a broken state, `Greenboot` scripts
+will restore the previously saved backup, and perform an atomic rollback to the N-1 state.
+
+### API Extensions
+
+None
+
+### Implementation Details/Notes/Constraints [Optional]
+
+#### rpm-ostree Configuration
+
+`rpm-ostree` provides a configuration option to check and stage new updates.
+The file `/etc/rpm-ostreed.conf` provides `AutomaticUpdatePolicy` which defaults to `none`.
+
+This option needs to be set to `stage` which downloads and unpacks the update performing
+any package layering. Only a small amount of work is left to be performed at shutdown time
+via the `ostree-finalize-staged.service` systemd unit.
+
+Furthermore, in order to apply this policy, the `rpm-ostreed-automatic.service` systemd
+service must be enabled.
```

Comment:
Can we add references to the `ostree` docs describing this functionality?

Reply:
I will link the following documentation:

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/composing_installing_and_managing_rhel_for_edge_images/managing-rhel-for-edge-images_composing-installing-managing-rhel-for-edge-images#enabling-rhel-for-edge-automatic-download-and-staging-of-updates_managing-rhel-for-edge-images

Thanks @ggiguash